extract_timeseries_tar.py

```diff
@@ -18,68 +18,162 @@
+def fetch_atlas_path(atlas_name, description_keywords, atlas_path, template):
+    """
+    Generate a dictionary containing parameters for TemplateFlow quiery.
+
+    Parameters
+    ----------
+    atlas_name : str
+        Atlas name. Must be a key in ATLAS_METADATA.
+
+    description_keywords : dict
+        Keys and values to fill in description_pattern.
+        For valid keys check relevant ATLAS_METADATA[atlas_name]['description_pattern'].
+
+    atlas_path : str
+        Path to self-defined atlas collection in TemplateFlow competable naming convention.
+
+    template : str
+        TemplateFlow template name.
+
+    Return
+    ------
+    sklearn.utils.Bunch
+        Containing the following fields:
+
+        maps : str
+            Path to atlas map.
+
+        labels : pandas.DataFrame
+            The corresponding pandas dataframe of the atlas
+
+        type : str
+            'dseg' (for NiftiLabelsMasker) or 'probseg' (for NiftiMapsMasker)
+    """
+
+    cur_atlas_meta = ATLAS_METADATA[atlas_name]
+    if cur_atlas_meta['source'] != "templateflow":
+        templateflow.conf.TF_HOME = pathlib.Path(atlas_path)
+        templateflow.conf.init_layout()
+        templateflow.conf.update(local=True)
+    img_parameters = generate_templateflow_parameters(cur_atlas_meta, "atlas", description_keywords)
+    label_parameters = generate_templateflow_parameters(cur_atlas_meta, "label", description_keywords)
+
+    img_path = str(templateflow.api.get(template, **img_parameters))
+    label_path = str(templateflow.api.get(template, **label_parameters))
     labels = pd.read_csv(label_path, delimiter="\t")
-    labels = (labels['Region'].astype(str) + ". " +
-              labels['Difumo_names']).values.tolist()
-
-    return sklearn.utils.Bunch(maps=img_path, labels=labels)
-
-
-def create_atlas_masker(atlas_name, atlas_path, dimension, resolution, nilearn_cache=""):
-    """Create masker of all dimensions and resolutions given metadata."""
-    if atlas_name not in ATLAS_METADATA.keys():
-        raise ValueError("{} not defined!".format(atlas_name))
-    curr_atlas = ATLAS_METADATA[atlas_name]
-    curr_atlas['name'] = atlas_name
-
-    atlas = eval(curr_atlas['fetcher'].format(
-        atlas_path=atlas_path, dimension=dimension, resolution=resolution))
-    if curr_atlas['type'] == "static":
-        masker = nilearn.input_data.NiftiLabelsMasker(
-            atlas.maps, detrend=True)
-    elif curr_atlas['type'] == "dynamic":
-        masker = nilearn.input_data.NiftiMapsMasker(
-            atlas.maps, detrend=True)
+    # labels = (labels['Region'].astype(str) + ". " +
+    #           labels['Difumo_names']).values.tolist()
+    atlas_type = img_path.split('_')[-1].split('.nii.gz')[0]
+    return sklearn.utils.Bunch(maps=img_path, labels=labels, type=atlas_type)
+
+
+def generate_templateflow_parameters(cur_atlas_meta, file_type, resolution, description_keywords):
```

Comment:
resolution parameter is not used, I imagine that this was an error. Already fixed in my version